src/query/server/server.go

```diff
@@ -337,14 +380,9 @@ func newStorages(
 			server.GracefulStop()
 		}
 
-		if remotes := cfg.RPC.RemoteListenAddresses; len(remotes) > 0 {
-			client, err := tsdbRemote.NewGrpcClient(remotes)
-			if err != nil {
-				logger.Fatal("unable to start remote clients for addresses", zap.Any("error", err))
-			}
-
-			stores = append(stores, remote.NewStorage(client))
-			remoteEnabled = true
+		if remoteStorage, enabled := remoteClient(logger, cfg); remoteEnabled {
```

Comment:
This seems odd. Should this be `if remoteStorage, enabled := remoteClient(logger, cfg); enabled {` ?